### PR TITLE
Introduce size check for render cache

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: #402 

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Introduce size check for render cache.</changelog>`.

### Summary of changes

Introduce cache size check when setting render cache.

### CODE REVIEW

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt [hunk 1]
```diff
@@ -1145,11 +1145,22 @@ class MapboxMap internal constructor(
...
    */
   @MapboxExperimental
   fun setRenderCacheOptions(options: RenderCacheOptions) {
-    nativeMapWeakRef.call { this.renderCacheOptions = options }
+    options.size?.let { size ->
+      nativeMapWeakRef.call {
+        this.renderCacheOptions = if (size >= 0)
+          options
+        else
+          RenderCacheOptions.Builder().setDisabled().build()
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:1159]: I think we should throw exceptions when the render cache is set to less than 0.

Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:1159]: In that case should we also throw an exception when `size` is not specified explicitly when building an object? IMHO it does not matter much until we have it all documented. Having `size == null || size < 0` ~ `cache disabled` seems pretty consistent to me.

Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:1159]: BTW I can still set cache to disabled if size < 0 but also log an error message @pengdev WDYT?

Comment (pengdev) [reply sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:1159]: IMO negative value is just wrong, user should never expect setting the size to a negative value should be permitted in any cases, I think we should throw illegal argument here.

Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/MapboxMap.kt]: You convinced me 👍

File: sdk/src/androidTest/java/com/mapbox/maps/RenderCacheTest.kt [hunk 2]
```diff
@@ -87,14 +87,30 @@ class RenderCacheTest {
...
-  @Ignore("TODO Uncomment after https://github.com/mapbox/mapbox-maps-android/issues/402")
   fun setInvalidRenderCache() {
     rule.scenario.onActivity {
       it.runOnUiThread {
-        val customSize = -1
-        mapView.getMapboxMap().setRenderCacheOptions(RenderCacheOptions.Builder().size(customSize).build())
+        val invalidSize = -1
+        var caughtException = false
+        try {
+          mapView.getMapboxMap().setRenderCacheOptions(RenderCacheOptions.Builder().size(invalidSize).build())
```
Comment (pengdev) [inline sdk/src/androidTest/java/com/mapbox/maps/RenderCacheTest.kt:96]: nit: we can use `@Test(expected = IllegalArgumentException.class)` to verify exceptions

**Review by pengdev — APPROVED** [review]
LGTM with nit

File: sdk/src/androidTest/java/com/mapbox/maps/RenderCacheTest.kt (hunk 2 above)
Comment (kiryldz) [reply sdk/src/androidTest/java/com/mapbox/maps/RenderCacheTest.kt]: Thanks, for some reason I thought it's not applicable for those tests 🤦

**Pull request merged**